sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90

```diff
@@ -101,109 +104,105 @@ subroutine eigensym2d(em,vv,oo,vvd,ood,ff)!                        [eigensym2]
 enddo
 end subroutine eigensym2d
 
-!=============================================================================
+!> Get the inverse of a 2*2 matrix (need not be symmetric in this case).
+!! @author R. J. Purser
+!! @param em 2*2 matrix
+!! @param z inverse of a 2*2 matrix
 subroutine invsym2(em,z)!                                            [invsym2]
-!=============================================================================
-! Get the inverse of a 2*2 matrix (need not be symmetric in this case).
-!=============================================================================
 implicit none
 real(dp),dimension(2,2),intent(in ):: em
 real(dp),dimension(2,2),intent(out):: z
-!-----------------------------------------------------------------------------
 real(dp):: detem
-!=============================================================================
 z(1,1)=em(2,2); z(2,1)=-em(2,1); z(1,2)=-em(1,2); z(2,2)=em(1,1)
 detem=em(1,1)*em(2,2)-em(2,1)*em(1,2)
 z=z/detem
 end subroutine invsym2
-!=============================================================================
+
+!> Get the inverse, z,of a 2*2 symmetric matrix, em, and its derivative, zd,
+!! with respect to symmetric variations of its components. I.e., for a
+!! symmetric infinitesimal change, delta_em, in em, the resulting
+!1 infinitesimal change in z would be:
+!! delta_z(i,j) = matmul(zd(i,j,:,:),delta_em)
+!! @author R. J. Purser
+!! @param z inverse of a 2*2 symmetric matrix
+!! @param em 2*2 symmetric matrix
+!! @param zd derivative of the 2*2 symmetric matrix
 subroutine invsym2d(em,z,zd)!                                        [invsym2]
-!=============================================================================
-! Get the inverse, z,of a 2*2 symmetric matrix, em, and its derivative, zd, 
-! with respect to symmetric variations of its components. I.e., for a 
-! symmetric infinitesimal change, delta_em, in em, the resulting
-! infinitesimal change in z would be:
-! delta_z(i,j) = matmul(zd(i,j,:,:),delta_em)
-!=============================================================================
 implicit none
 real(dp),dimension(2,2)    ,intent(in ):: em
 real(dp),dimension(2,2)    ,intent(out):: z
 real(dp),dimension(2,2,2,2),intent(out):: zd
-!-----------------------------------------------------------------------------
 integer(spi):: k,l
-!=============================================================================
 call invsym2(em,z)
 call id2222(zd)
 do l=1,2; do k=1,2
    zd(:,:,k,l)=-matmul(matmul(z,zd(:,:,k,l)),z)
 enddo;    enddo
 end subroutine invsym2d
 
-!=============================================================================
+!> Get the sqrt of a symmetric positive-definite 2*2 matrix
+!! @author R. J. Purser
```

Comment:
Please put the author tag at the bottom of the docblock in all cases.

Reply:
> Please put the author tag at the bottom of the docblock in all cases.

@edwardhartnett Please explain a bit more in detail on this comment. 
I inserted @author tag in subroutine sqrtsym2. What else should I do?

Reply:
If there is a date tag, I usually put it on the same line as the author:

`!! @author Freddy Fortran @date 12-Dec-1981
`
(The format of the date doesn't matter, can be 12/12/81, or just December 1981, or any reasonable date format.)

If there is no date in the existing documentaiton, I don't worry about it.

If there is no author, I look in the GitHub "blame" screen to show who checked in the code, or I just assume whoever wrote the file also wrote all the functions in the file.